packages/integration-karma/test/light-dom/multiple-templates/index.spec.js

```diff
@@ -0,0 +1,32 @@
+import { createElement, setFeatureFlagForTest } from 'lwc';
+
+import Multi from 'x/multi';
+
+describe('multiple templates', () => {
+    beforeEach(() => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', true);
+    });
+
+    afterEach(() => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', false);
+    });
+
+    it('can render multiple templates with different styles', () => {
+        const element = createElement('x-multi', { is: Multi });
+
+        document.body.appendChild(element);
+
+        expect(element.querySelector('div').textContent).toEqual('a');
+        expect(getComputedStyle(element.querySelector('div')).color).toEqual('rgb(233, 150, 122)');
+        expect(getComputedStyle(element.querySelector('div')).marginLeft).toEqual('0px');
+        element.querySelector('div').setAttribute('foo', '');
+        element.next();
+        return new Promise((resolve) => requestAnimationFrame(() => resolve())).then(() => {
```

Comment:
A single microtask is needed here, no need to wait for a full RAF to get the content updated.

```suggestion
        return Promise.resolve().then(() => {
```